node/router/config_submitter_test.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/hyperledger/fabric-x-orderer/testutil"
+	"github.com/hyperledger/fabric-x-orderer/testutil/tx"
+	"github.com/stretchr/testify/require"
+)
+
+type configSubmitTestSetup struct {
+	stubConsenter   *stubConsenter
+	configSubmitter *configSubmitter
+}
+
+func (s configSubmitTestSetup) Start() {
+	s.stubConsenter.Start()
+	s.configSubmitter.Start()
+}
+
+func (s configSubmitTestSetup) Stop() {
+	s.stubConsenter.Stop()
+	s.configSubmitter.Stop()
+}
+
+func createconfigSubmitTestSetup(t *testing.T) configSubmitTestSetup {
+	logger := testutil.CreateLogger(t, 0)
+	ca, err := tlsgen.NewCA()
+	require.NoError(t, err)
+
+	stubConsenter := NewStubConsenter(t, ca, types.PartyID(1), logger)
+
+	ckp, err := ca.NewServerCertKeyPair("127.0.0.1")
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	configSubmitter := configSubmitter{
+		consensusEndpoint:     stubConsenter.GetConsenterEndpoint(),
+		consensusRootCAs:      [][]byte{ca.CertBytes()},
+		tlsCert:               ckp.Cert,
+		tlsKey:                ckp.Key,
+		logger:                logger,
+		configRequestsChannel: make(chan *TrackedRequest, 100),
+		doneChannel:           make(chan struct{}),
+		ctx:                   ctx,
+		cancelFunc:            cancel,
+	}
+	return configSubmitTestSetup{configSubmitter: &configSubmitter, stubConsenter: &stubConsenter}
+}
+
+func TestConfigSubmitterForward(t *testing.T) {
+	setup := createconfigSubmitTestSetup(t)
+	setup.Start()
+	defer setup.Stop()
+	configSubmitter := setup.configSubmitter
+	stubConsenter := setup.stubConsenter
+
+	req := tx.CreateStructuredRequest([]byte("data"))
+	feedbackChan := make(chan Response, 1)
+	configSubmitter.Forward(&TrackedRequest{request: req, responses: feedbackChan})
+
+	resp := <-feedbackChan
+	require.Equal(t, resp.SubmitResponse.Error, "dummy submit config")
+
+	require.Eventually(t, func() bool {
+		return stubConsenter.ReceivedMessageCount() == uint32(1)
+	}, 10*time.Second, 10*time.Millisecond)
+}
+
+func TestConfigSubmitterMultipleForward(t *testing.T) {
+	setup := createconfigSubmitTestSetup(t)
+	setup.Start()
+	defer setup.Stop()
+	configSubmitter := setup.configSubmitter
+	stubConsenter := setup.stubConsenter
+
+	numOfRequests := 10
+	for i := 0; i < numOfRequests; i++ {
+		req := tx.CreateStructuredRequest([]byte("data"))
+		feedbackChan := make(chan Response, 1)
+		configSubmitter.Forward(&TrackedRequest{request: req, responses: feedbackChan})
+
+		resp := <-feedbackChan
+		require.Equal(t, resp.SubmitResponse.Error, "dummy submit config")
+	}
+
+	require.Eventually(t, func() bool {
+		return stubConsenter.ReceivedMessageCount() == uint32(numOfRequests)
+	}, 10*time.Second, 10*time.Millisecond)
+}
+
+func TestForwardWithConsensusRestart(t *testing.T) {
+	setup := createconfigSubmitTestSetup(t)
+	setup.Start()
+	defer setup.Stop()
+
+	configSubmitter := setup.configSubmitter
+	stubConsenter := setup.stubConsenter
+
+	// wait until consenter is up
```

Comment:
just submit once and make sure it made it instead of waiting here
you don't wait with other tests